### ISSUE TITLE
.card flex rules break in Internet Explorer 10-11

### ISSUE THREAD

**Issue by HowlingEverett** [twbs/bootstrap#18569]
The card flex shorthand rule is:

```css
.card {
   flex: 1 0 0;
}
```

In Internet Explorer, the `flex-basis` rule of 0 causes the cards to ignore the `flex-grow` rule and size based on horizontal content-width.

Replacing the rule with:

```css
.card {
  flex: 1 0 1px;
}
```

fixes the issue in IE and still works as expected in other browsers (cards grow to fill the container but maintain equal width in relation to each other.)

**Comment by petetnt** [comment]
Related issue here: #17310

**Comment by cvrebert** [comment]
This is an instance of [Flexbug #4](https://github.com/philipwalton/flexbugs#4-flex-shorthand-declarations-with-unitless-flex-basis-values-are-ignored).
The problem happens when using a unitless `0` value. The proper workaround is to include a unit,
i.e. `flex: 1 0 0px;` or `flex: 1 0 0%;`
We'll also need to check that clean-css preserves the unit here when minifying.

**Comment by cvrebert** [comment]
Looks like clean-css has a special case to handle this properly.

**Comment by cvrebert** [comment]
Perhaps we should integrate https://github.com/luisrudge/postcss-flexbugs-fixes into the build.

**Comment by cvrebert** [comment]
Currently [this CSSComb setting](https://github.com/twbs/bootstrap/blob/d5ec7e3f502446c13623010a4a50866d3c29be54/scss/.csscomb.json#L20) would defeat any potential fix.

**Comment by cvrebert** [comment]
Perhaps we should migrate from CSSComb to https://github.com/hudochenkov/postcss-sorting

**Comment by petetnt** [comment]
Heh, @cvrebert, nice timing, I just integrated `postcss-flexbugs-fixes` into our own project, only to notice that [the CSSComb setting](https://github.com/twbs/bootstrap/blob/d5ec7e3f502446c13623010a4a50866d3c29be54/scss/.csscomb.json#L20) defeats it... :hammer:

**Comment by cvrebert** [comment]
CSSComb has been killed in f25f9bceb1924b54f5743a238ca04c162731136d, so we should be able to move ahead with this.

**Comment by cvrebert** [comment]
Fixed by #19109.

**Comment by obinnangini** [comment]
I just ran into this issue again while compatibility testing in Internet Explorer 11.
I'm importing ``bootstrap@4.0.0-alpha.6`` ``.scss`` files via npm and compiling to ``.css`` using webpack. 
As a result I'm not getting the fixes the ``postcss-flexbugs-fixes`` library applies to the compiled CSS. 
I searched the ``scss`` directory and there are only 2 occurences of the ``flex:`` property without a measurement unit, both of which are in the ``_card.scss`` file. 
Is it possible to add the measurement in the scss files so the fix is propagated to people who use the scss files and have to support Internet Explorer?
I would prefer to not have to pull in the ``postcss-flexbugs-fixes`` library as a dependency and add it to my build process. 
It seems that anyone compiling the ``.scss`` files to ``.css`` would have to do this, which seems counter-intuitive.

Thanks for building a great style library! Really enjoying how easy it is to use.
